docs/developer-experience/readme.md

```diff
@@ -0,0 +1,231 @@
+
+Thus, the ideal scenario could consist:
+
+- A set of one or more services, every one of them with its own business needs. Following the micro-services architecture you deploy each of those services independently.
+- Those same set of services should share a common set of technical requirements in terms of security, token management, external calls management, error management, logging, etc.
+
+### What about other scenarios?
+
+Depending on the scenario is up to you to decide whether bauta.js is useful to you or not. We hope that this document can provide information to easily assist you in that decision.
+
+For example, if you are exposing a GraphQL server then you can decide to use directly a GraphQL server like Apollo or Mercurius and you may think that you do not need bauta.js [^1].
+
+Another case would be if your REST API is an entire monolith. Is it worth it to learn bauta.js if you are maintaining an already big monolith? As usual, it depends, but the advantage of having your monolith in bauta.js is that it should be easier to separate it in different micro-services if you have the need to do so in the future. 
+
+## The key concepts of bauta.js
+
+### StepFunction: the working unit of bauta.js
+
+StepFunction is a function in bauta.js that conforms to the following signature:
+
+- it has three input parameters: prev, context and bauta.js instance itself.
+- it returns a value or a promise that would eventuall return a value.
+
+#### Why it is so important?
+
+Because not only the application code uses this as a type. Bauta.js itself uses this signature for its internal execution. Even bauta.js more advanced abstractions are in turn or return a StepFunction. And finally, because almost all the logic defined in terms of endpoints are a combination of one or more StepFunctions (check pipeline for more details).
+
+One could Think about the code of a bauta.js application as a bunch of StepFunctions that are run one after another or in parallel and whose results are passed from one to the next defined in your code. 
+
+#### What are the main parts of an StepFunction?
+
+These are three input parameters of an StepFunction:
+
+- prev: is the previous value. this always take the value (or values) returned by the previous StepFunction. This value can be undefined, usually because is the the first StepFunction and there is none previously.
+
+- context: this is the context of the request being processed. You can think of this parameter as the session of the request [^2].
+
+- bautaJSInstance: this object represents the bauta.js server instance itself. The most used field of this object is called staticConfig and has all the variables loaded by bauta.js at startup time.
```

Comment:
```suggestion
These are three input parameters of a StepFunction:

- prev: is the previous value. This always takes the value (or values) returned by the previous `StepFunction`. This value can be undefined, usually because it is the first StepFunction and there is none previously.

- context: this is the context of the request being processed. You can think of this parameter as the session of the request [^2].

- bautaJSInstance: this object represents the bauta.js server instance itself. The most used field of this object is called staticConfig and has all the variables loaded by bauta.js at startup time.
```